src/rand_impls.rs

```diff
@@ -176,6 +201,129 @@ impl Rand for bool {
     }
 }
 
+impl Rand for Ordering {
+    #[inline]
+    fn rand<R: Rng>(rng: &mut R) -> Ordering {
+        match rng.gen_range(0, 3) {
+            0 => Ordering::Less,
+            1 => Ordering::Equal,
+            2 => Ordering::Greater,
+            _ => unreachable!(),
+        }
+    }
+}
+
+impl Rand for Duration {
+    #[inline]
+    fn rand<R: Rng>(rng: &mut R) -> Duration {
+        Duration::new(rng.gen(), rng.gen_range(0, 1_000_000_000))
+    }
+}
+
+impl Rand for RangeFull {
+    #[inline]
+    fn rand<R: Rng>(rng: &mut R) -> RangeFull {
+        ..
+    }
+}
+
+impl<T: Rand + PartialOrd> Rand for Range<T> {
+    #[inline]
+    fn rand<R: Rng>(rng: &mut R) -> Range<T> {
+        let start = rng.gen();
+        let end = rng.gen();
+        if end < start {
+            Range { start: end, end: start }
+        } else {
+            Range { start: start, end: end }
+        }
```

Comment:
@nagisa Wouldn't this cause issues with `Range<usize>` or similar types where (correct me if I'm wrong) `start` is expected to be less than `end`?